clients/unshaded/src/main/java/tachyon/client/lineage/LineageFileOutStream.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.lineage;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.annotation.PublicApi;
+import tachyon.client.UnderStorageType;
+import tachyon.client.block.BufferedBlockOutStream;
+import tachyon.client.file.FileOutStream;
+import tachyon.client.file.options.OutStreamOptions;
+
+@PublicApi
+public class LineageFileOutStream extends FileOutStream {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private LineageContext mContext;
+
+  public LineageFileOutStream(long fileId, OutStreamOptions options) throws IOException {
+    super(fileId, updateOutStreamOptions(options));
+    mContext = LineageContext.INSTANCE;
+  }
+
+  private static OutStreamOptions updateOutStreamOptions(OutStreamOptions options) {
+    // change the under storage type to async
+    OutStreamOptions.Builder builder = new OutStreamOptions.Builder(options);
```

Comment:
Since this convenience method is only used here, it may make sense to make it a convenience method in this class instead of adding it to the outstreamoptions.